inc/class-rest-api.php

```diff
@@ -886,11 +905,79 @@ public function update_item_permissions_check( $request ) {
 	public function update_item( $request ) {
 		$this->add_sst_request_filters();
 
-		$data = [];
+		$this->created_refs     = [];
+		$this->errors           = [];
+		$this->response_objects = [];
+
+		$post_id = $request->get_param( 'id' );
 
-		$response = rest_ensure_response( $data );
+		// Validate the post to be updated.
+		$prepared_post = $this->prepare_item_for_database( $request );
+		if ( is_wp_error( $prepared_post ) ) {
+			return $prepared_post;
+		}
 
-		return $response;
+		// Defer to the core REST API endpoint to update the post.
+		$post_type_obj = get_post_type_object( $request['type'] );
+		$response      = $this->dispatch_request(
+			'PUT',
+			sprintf(
+				'/wp/v2/%s/%d',
+				( $post_type_obj->rest_base ?: $post_type_obj->name ),
+				$post_id
+			),
+			[ 'body' => $prepared_post ]
+		);
```

Comment:
Although unusual, it might be worth keeping in mind that this request would not work as expected for post types that set a `rest_controller_class` outside of the `WP_REST_Posts_Controller` hierarchy.

Reply:
I'm OK with that